### ISSUE TITLE
Recipes for Debian emacs-goodies-el?

### ISSUE THREAD

**Issue by DarwinAwardWinner** [melpa/melpa#5449]
There are a couple of useful tools in the Debian package emacs-goodies-el that might be of interest outside Debian systems. For example, R packages use the Debian control file format, which has a major mode in this package. Hence, it would be nice to have MELPA packages for some of these elisp files. I don't know the details, but at some point I guess Debian migrated to git, which means that there is now an official SCM repo for this package here: https://salsa.debian.org/debian/emacs-goodies-el. Hence, the criteria for making a MELPA recipe are technicaly satisfied.

However, things might be complicated by the fact that the debian package build process involves applying a series of patches (see the `debian/patches` subdirectory of the repo). I don't know if these patches are fixes specific to making these packages work in Debian, whether they are always meant to be applied, or whether it is a mix of the two. In any case, I don't think there's a mechanism for applying patches in the MELPA build process.

So, do you think it's worth looking into making MELPA packages for some of the files in emacs-goodies-el, or is it not practical?

**Comment by purcell** [comment]
Interesting. If there are well-formed elisp packages there which have no other upstream, this would be an acceptable source for them. But yes, we wouldn't be able to apply patches, so you'd have to look at what happens to specific files. Where patches are getting applied by the debian build process, presumably those patches are debian-specific and/or there is still an upstream which can't/won't accept those patches directly.

**Comment by DarwinAwardWinner** [comment]
So I got some more time to look into it and the main set of scripts that are potentially useful and have no other upstream are the ones in [dpkg-dev-el](https://salsa.debian.org/debian/emacs-goodies-el/tree/master/elisp/dpkg-dev-el). There don't appear to be any patches affecting these files, as one would expect since this is already the upstream source for them. I've written recipes for the ones I want, and it's as trivial as you would hope: https://github.com/DarwinAwardWinner/dotemacs/blob/master/config.org#dpkg-dev-el

If you'd like, I can submit a pull request with a recipe for each of the scripts in this directory of the repo.

**Comment by purcell** [comment]
> If you'd like, I can submit a pull request with a recipe for each of the scripts in this directory of the repo.

Yup, sounds good, thanks.

**Comment by DarwinAwardWinner** [comment]
Ok, package-lint has some complaints, which may or may not be worth the effort to address, since I have no idea how receptive the Debian maintainers will be toward changes whose only purpose is to accommodate a non-dpkg packaging system. 

So, how many of the below are showstoppers? (This is just for one of the files, but the rest will be similar.)

```
1:1: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
1:1: error: Package should have a Homepage or URL header.
1:1: error: package.el cannot parse this buffer: Package lacks a "Version" or "Package-Version" header
24:0: error: Package should have a non-empty ;;; Commentary section.
498:1: error: "match-string-no-properties" doesn't start with package's prefix "debian-changelog".
1739:1: error: "imenu--create-debian-changelog-index" doesn't start with package's prefix "debian-changelog".
```

**Comment by purcell** [comment]
The only errors I'd really worry about here would be the last 2, so it'd be worth sending a patch upstream if it's not too painful to do so. Ultimately I'd be willing to merge recipes for these even without the above being fixed.

**Comment by DarwinAwardWinner** [comment]
FWIW, the error about `match-string-no-properties` is spurious. It's conditionally defining it only on very old Emacsen (i.e. XEmace 21) as a compatibility shim. (I edited my previous post to say this, but you might not have seen the edit if you were reading via email.) In any case, I'll submit pull requests for each of these soon.

**Comment by bremner** [comment]
Just FYI, we're in the process of breaking up emacs-goodies-el in Debian, and would welcome people to take over upstream for any bits that are currently without a viable upstream. It's true we probably are not super interested in minor fixes, but if you discover some serious user harm then we'll have to deal with it.

**Comment by tarsius** [comment]
@bremner Any update on this?

**Comment by bremner** [comment]
Jonas Bernoulli <notifications@github.com> writes:

> @bremner Any update on this?
>

emacs-goodies-el is mostly an empty-shell in Debian
testing/unstable. There's two packages left, ff-paths.el and
projects.el. projects.el will probably be just dropped. ff-paths.el
might become a seperate package, that's a bit up to the author (Peter
Galbraith).

The debian specific stuff has been split off into it's own debian
packages, debian-el and dpkg-dev-el.  I guess if you want to package
those in melpa, it's up to you. I expect Debian to continue being
upstream for those for the medium term.  OTOH, there are some
dependencies on e.g. dpkg and reportbug that might make a melpa version
less useful.

d

**Comment by tarsius** [comment]
I had another look at `emacs-goodies-el` and can't quite tell what it does -- it appears it essentially provides to (unrelated?) libraries `project.el` and`ff-paths.el`. The former you said will likely be removed and the latter appears to be obsolete too, it contains:

> I suggest that you use ffap.el by Michelangelo Grigni <mic@cs.ucsd.edu>, now part of GNU Emacs.

So I am going to ignore https://salsa.debian.org/emacsen-team/emacs-goodies-el and concentrate my investigations on https://salsa.debian.org/emacsen-team/debian-el.

**Comment by tarsius** [comment]
`debian-el` is mostly fine. The contained libraries could be used independently, but because they are related and because they are tied together by the presence of the `debian-el.el` library that is okay and I think we should this as a single package `debian-el` to Melpa.

However `debian-el.el` needs some work. It needs a library commentary and a license statement. I have added this with the last two commits at https://github.com/emacsjanitors/debian-el. Please review and merge. Once this is done this package can be added to Melpa. There are a few other commits, but they are just cosmetic.

**Comment by bremner** [comment]
Jonas Bernoulli <notifications@github.com> writes:

> `debian-el` is mostly fine. The contained libraries could be used independently, but because they are related and because they are tied together by the presence of the `debian-el.el` library that is okay and I think we should this as a single package `debian-el` to Melpa.
>
> However `debian-el.el` needs some work. It needs a library commentary
> and a license statement. I have added this with the last two commits
> at https://github.com/emacsjanitors/debian-el. Please review and
> merge. Once this is done this package can be added to Melpa. There are
> a few other commits, but they are just cosmetic.

Pushed to salsa.

I haven't tagged a new release yet.

d

**Comment by tarsius** [comment]
Thanks!

Could you do the same for https://salsa.debian.org/emacsen-team/dpkg-dev-el.git, adding the license statement and a better commentary to `dpkg-dev-el.el`?

**Comment by bremner** [comment]
Package: dpkg-dev-el
Version: 37.4
Severity: wishlist
Tags: newcomer

Jonas Bernoulli asks for a license statement and commentary section in
dpkg-dev-el.el. This is apparently needed by the melpa machinery,


-- System Information:
Debian Release: buster/sid
  APT prefers testing
  APT policy: (500, 'testing')
Architecture: amd64 (x86_64)
Foreign Architectures: i386

Kernel: Linux 4.18.0-1-amd64 (SMP w/8 CPU cores)
Locale: LANG=en_CA.UTF-8, LC_CTYPE=en_CA.UTF-8 (charmap=UTF-8), LANGUAGE=en_CA:en (charmap=UTF-8)
Shell: /bin/sh linked to /bin/dash
Init: systemd (via /run/systemd/system)
LSM: AppArmor: enabled

Versions of packages dpkg-dev-el depends on:
ii  elpa-dpkg-dev-el  37.4

dpkg-dev-el recommends no packages.

dpkg-dev-el suggests no packages.

-- no debconf information

**Comment by tarsius** [comment]
This was done a few days ago. I've added the recipes now.
